File: app/src/main/kotlin/com/mattdixon/jobjar/util/RecurrenceFormat.kt
package com.mattdixon.jobjar.util

private const val DAY_MILLIS = 24L * 60 * 60 * 1000

/** Friendly label for a "repeats every N days" interval. */
fun formatRecurrenceInterval(days: Int): String = when (days) {
    1 -> "Daily"
    7 -> "Weekly"
    14 -> "Biweekly"
    30 -> "Monthly"
    else -> "Every $days days"
}

/**
 * "Due now" if [nextDueAt] is null (never completed yet) or already passed, otherwise how far
 * out the next occurrence is. Rounds up so "due tomorrow morning" doesn't read as "in 0 days".
 */
fun formatDueStatus(nextDueAt: Long?, now: Long = System.currentTimeMillis()): String {
    if (nextDueAt == null || nextDueAt <= now) return "Due now"
    val daysAway = ((nextDueAt - now) + DAY_MILLIS - 1) / DAY_MILLIS
    return if (daysAway == 1L) "Next: tomorrow" else "Next: in $daysAway days"
}

File: ios/JobJar/JobJar/Util/RecurrenceFormat.swift
import Foundation

private let dayInterval: TimeInterval = 24 * 60 * 60

/// Friendly label for a "repeats every N days" interval.
func formatRecurrenceInterval(_ days: Int) -> String {
    switch days {
    case 1: return "Daily"
    case 7: return "Weekly"
    case 14: return "Biweekly"
    case 30: return "Monthly"
    default: return "Every \(days) days"
    }
}

/// "Due now" if nextDueAt is nil (never completed yet) or already passed, otherwise how far
/// out the next occurrence is. Rounds up so "due tomorrow morning" doesn't read as "in 0 days".
func formatDueStatus(_ nextDueAt: Date?, now: Date = .now) -> String {
    guard let nextDueAt, nextDueAt > now else { return "Due now" }
    let daysAway = Int(ceil(nextDueAt.timeIntervalSince(now) / dayInterval))
    return daysAway == 1 ? "Next: tomorrow" : "Next: in \(daysAway) days"
}
